File: solution/2_Decodering/2_Decodering/LengteDecoderingAlgoritmeTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using decodering;

namespace test
{
    [TestClass]
    public class LengteDecoderingAlgoritmeTest
    {
        private LengteDecoderingAlgoritme lengteDecodering;

        [TestInitialize]
        public void setup() {
		    lengteDecodering = new LengteDecoderingAlgoritme();
	    }

        [TestMethod]
        public void legeString() {
		    Assert.AreEqual("", lengteDecodering.decodeer(""));
	    }

	    [TestMethod]
	    public void geenDecoderingNodig() {
		    Assert.AreEqual("abc", lengteDecodering.decodeer("abc"));
	    }

	    [TestMethod]
	    public void decodeerVolledigeString() {
		    Assert.AreEqual("aaa", lengteDecodering.decodeer("3a"));
	    }

	    [TestMethod]
	    public void decodeerEersteLetters() {
		    Assert.AreEqual("aaabc", lengteDecodering.decodeer("3abc"));
	    }

	    [TestMethod]
	    public void decodeerLettersInMidden() {
		    Assert.AreEqual("abbbbc", lengteDecodering.decodeer("a4bc"));
	    }

	    [TestMethod]
	    public void decodeerLettersOpEinde() {
		    Assert.AreEqual("abcc", lengteDecodering.decodeer("ab2c"));
	    }

        [TestMethod]
	    public void meerdereAaneenschakelingenVanHetzelfdeKarakter() {
		    Assert.AreEqual("aaaabaaaaaaaaaaaaaacaaaaa", lengteDecodering.decodeer("4ab14ac5a"));
	    }
    }
}

File: workspace/2_Decodering/test/decodering/LengteDecoderingAlgoritmeTest.java
package decodering;

import static junit.framework.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;


public class LengteDecoderingAlgoritmeTest {

	private LengteDecoderingAlgoritme lengteDecodering;

	@Before
	public void setup() throws Exception {
		lengteDecodering = new LengteDecoderingAlgoritme();
	}

	@Test
	public void legeString() {
		assertEquals("", lengteDecodering.decodeer(""));
	}

	@Test
	public void geenDecoderingNodig() {
		assertEquals("abc", lengteDecodering.decodeer("abc"));
	}

	@Test
	public void decodeerVolledigeString() {
		assertEquals("aaa", lengteDecodering.decodeer("3a"));
	}

	@Test
	public void decodeerEersteLetters() {
		assertEquals("aaabc", lengteDecodering.decodeer("3abc"));
	}

	@Test
	public void decodeerLettersInMidden() {
		assertEquals("abbbbc", lengteDecodering.decodeer("a4bc"));
	}

	@Test
	public void decodeerLettersOpEinde() {
		assertEquals("abcc", lengteDecodering.decodeer("ab2c"));
	}

	@Test
	public void meerdereAaneenschakelingenVanHetzelfdeKarakter() {
		assertEquals("aaaabaaaaaaaaaaaaaacaaaaa", lengteDecodering.decodeer("4ab14ac5a"));
	}

}
